server/build-plugins.js

```diff
@@ -62,24 +120,19 @@ const prepareModules = async (plugins = [], local = true) => {
         });
       }
       if (!local) {
-        // check if connected to internet
-        // if not, skip npm install
-        const EXTERNAL_URI = process.env.EXTERNAL_URI || 'npmjs.com';
-        await require('dns').lookup(EXTERNAL_URI, async err => {
-          if (err && err.code === 'ENOTFOUND')
-            logger.error("Can't reach npm servers. Skipping npm install");
-          else {
-            logger.info('Installing plugin packages...');
-            await execSync(
-              `npm install --no-save ${installPackages.join(' ')} --production`,
-              {
-                stdio: [0, 1, 2],
-                cwd: resolve(__dirname, '..')
-              }
-            );
-            logger.info('Done installing plugins');
-          }
-        });
+        // check if modules need to be installed by confirming
+        // if any plugins don't exist yet in our node_modules
+        const newModules = installPackages.some(
+          pkg => !fs.existsSync(resolve(modulesDirectory, pkg))
+        );
+
+        if (newModules) await installRemotePackages(installPackages);
```

Comment:
This is new and prevents `npm install` if the plugins all exist. If one doesn't exist then it will run `installRemotePackages`